File: C#2/Homeworks/06-Strings-and-Text-Processing/02ReverseString/ReverseString.cs
/*
•Write a program that reads a string, reverses it and prints the result on the console.

Input
•On the only line you will receive a string

Output
•Print the string in reverse on a single line

Constraints
•1 <= size of string <= 10000
•Time limit: 0.1s
•Memory limit: 16MB
 */
using System;
using System.Text;
class ReverseString
{
    static void Main()
     {
         string input = Console.ReadLine();
         string reversed = ReversingString(input);
         //string reversed = ReversingStringBuilder(input);

         Console.WriteLine(reversed);

     }

    static string ReversingString(string str)
    {
        string reversed = "";
        for (int i = str.Length -1;  i >=0;  i--)
        {
            reversed += str[i];
        }
        return reversed;
    }

    static string ReversingStringBuilder(string str)
    {
        StringBuilder sb = new StringBuilder();
        for (int i = str.Length - 1; i >= 0; i--)
        {
            sb.Append(str[i]);
        }
        return sb.ToString();
    }
}

File: JS/HW/Strings/reverse-string.js
// Write a JavaScript function that reverses a string.

// The input will consist of an array containing one string
// The output should be consisted of one line - the reversed string

function reverseString(args){
    let str = args[0],
        reversed ='';

for(let i = str.length-1; i>=0;i-=1){
        reversed += str[i];
    }
    console.log(reversed);
}

reverseString(['sample']);
reverseString(['qwertytrewq']);